lib/Result.hpp

```diff
@@ -0,0 +1,88 @@
+
+/*
+ * Result.hpp
+ *
+ *  Created on: 2021/09/25
+ *      Author: hommyo
+ *
+ * This cord is written by UTF-8
+ */
+
+#define RESULT_HPP_
+
+#define _USE_MATH_DEFINES
+
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
+#include <ctime>
+
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <sstream>
+#include <string>
+#include <regex>
+#include <functional>
+
+using namespace std;
+
+struct Result
+{
+protected:
+    double value;
+    vector<double> objective_function_value;
+    vector<double> design_variable;
+
+public:
+
+    Result(){}
+
+    Result& set_value(double input)
+    { 
+        value = input;
+        return *this; 
+    }
+
+    double get_value() const
+    { return value; }
+    vector<double> get_objective_function_value() const
+    { return objective_function_value; }
+    vector<double> get_design_variable() const
+    { return design_variable; }
+
+    //CSV出力用関数
+    void csv_result(const vector<Result>&);
+    void csv_statistics(const double);
+    void csv_objective_function_value(const vector<double>);
+    void csv_design_variable(const vector<double>);
```

Comment:
```suggestion
    void tprint_update_curve(const vector<double>);
    void tprint_update_variable(const vector<double>);
    void tprint_variables(const vector<double>);
```

- 引数がなぞ
- update_curveが更新曲線、update_valsが変数の更新経歴、variablesが最良変数
- tprint <--> text print, gprintと対比して

合わせて内部変数も変えた方がいいかも。
ここでは最適化一般の名称を選択したいと思ってます
←メタヒューリスティクスはフィルタ設計以外にも適用できるので

Reply:
あと出力ファイル名の引数も欲しい気がする

Reply:
内部変数を最適化一般の名称への変更というと、変数名にobjective_function_valueなどを用いないようにということでしょうか

Reply:
`objective_function_value`は一般的だけど、
コード中では長いから`value`に短縮で統一してる

`design variables`が変数(variables)でいいのに、
応用サイドなので**_設計_**変数となってしまっている